core/utils/ethabi.go

```diff
@@ -27,6 +29,39 @@ const (
 	FormatBool = "bool"
 )
 
+// GenericEncode eth encodes values based on the provided types
+func GenericEncode(types []string, values ...interface{}) ([]byte, error) {
+	if len(values) != len(types) {
+		return nil, errors.New("must include same number of values as types")
+	}
+	var args abi.Arguments
+	for _, t := range types {
+		ty, _ := abi.NewType(t, "", nil)
+		args = append(args, abi.Argument{Type: ty})
+	}
+	out, err := args.PackValues(values)
+	if err != nil {
+		return nil, err
+	}
+	return out, nil
+}
+
+func MustGenericEncode(types []string, values ...interface{}) []byte {
```

Comment:
Why don't we call `GenericEncode` and panic on error?

Reply:
I just ported this code, will refactor it as you suggest